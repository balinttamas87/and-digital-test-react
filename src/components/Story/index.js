import React from "react";
import PropTypes from "prop-types";

const Story = ({ title, score, by }) => (
  <div className="story">
    <div className="story-title-wrapper">
      <p className="story-title">{title}</p>
    </div>
    <div className="story-details-wrapper">
      <p className="story-details">{`${score} points by ${by}`}</p>
    </div>
  </div>
);

Story.propTypes = {
  title: PropTypes.string,
  score: PropTypes.number,
  by: PropTypes.string
};

export default Story;
