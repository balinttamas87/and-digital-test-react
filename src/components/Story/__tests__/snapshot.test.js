import React from "react";
import renderer from "react-test-renderer";
import Story from "..";

it("Renders <Story /> component and matches snapshot", () => {
  const component = renderer.create(
    <Story title="Title" score={100} by="author" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
