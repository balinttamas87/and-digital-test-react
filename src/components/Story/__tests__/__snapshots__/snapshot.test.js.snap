// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Story /> component and matches snapshot 1`] = `
<div
  className="story"
  onClick={[Function]}
>
  <div
    className="story-title-wrapper"
  >
    <p
      className="story-title"
    >
      Title
    </p>
  </div>
  <div
    className="story-details-wrapper"
  >
    <p
      className="story-details"
    >
      100 points by author
    </p>
  </div>
</div>
`;
